### ISSUE TITLE
Webpack 4 Migration

### ISSUE THREAD

**Issue by darylteo** [jeerbl/webfonts-loader#43]
```
    ERROR in ./node_modules/string-replace-webpack-plugin/loader.js?id=12twekiwixhc!./node_modules/css-loader!./node_modules/webfonts-loader?fileName=./fonts/[fontname].[ext]!./src/icons/listcorp.font.js
    Module build failed: TypeError: Cannot read property 'context' of undefined
        at webfontsGenerator (/Users/darylteo/workspaces/listcorp/dev/lca/frontend/node_modules/webfonts-loader/index.js:172:35)
        at /Users/darylteo/workspaces/listcorp/dev/lca/frontend/node_modules/webfonts-generator/src/index.js:105:4
        at _fulfilled (/Users/darylteo/workspaces/listcorp/dev/lca/frontend/node_modules/q/q.js:854:54)
        at self.promiseDispatch.done (/Users/darylteo/workspaces/listcorp/dev/lca/frontend/node_modules/q/q.js:883:30)
        at Promise.promise.promiseDispatch (/Users/darylteo/workspaces/listcorp/dev/lca/frontend/node_modules/q/q.js:816:13)
        at /Users/darylteo/workspaces/listcorp/dev/lca/frontend/node_modules/q/q.js:624:44
        at runSingle (/Users/darylteo/workspaces/listcorp/dev/lca/frontend/node_modules/q/q.js:137:13)
        at flush (/Users/darylteo/workspaces/listcorp/dev/lca/frontend/node_modules/q/q.js:125:13)
        at process._tickCallback (internal/process/next_tick.js:150:11)
```

--- 

https://medium.com/webpack/webpack-4-migration-guide-for-plugins-loaders-20a79b927202

> this.options and this.rootContext
webpack 3 already deprecated this.options in the loader context. webpack 4 removes it now.

> A lot of people are missing the this.options.context value now. It has been added as this.rootContext.

> Guideline: Loaders should receive all options via this.query. They should not use other ways to receive options, i. e. no property in webpack options, no environment variable.

--- 

[Example fix](https://github.com/bholloway/resolve-url-loader/pull/80/files)? Not sure. I'm not too familiar with Webpack loaders.

Thanks!

**Comment by frlinw** [comment]
- Remove ref to `this.options`
https://github.com/jeerbl/webfonts-loader/blob/master/index.js#L67
https://github.com/jeerbl/webfonts-loader/blob/master/index.js#L141
- Replace `this.options.context` by `this.rootContext` https://github.com/jeerbl/webfonts-loader/blob/master/index.js#L172

You think you can do it @darylteo ?

**Comment by jeerbl** [comment]
Hey people,

I'm opened to pull requests so if you know how to fix this, please do it ;)

Cheers
Jerome

**Comment by ptitdam2001** [comment]
The pull request is created, waiting for merge

**Comment by jeerbl** [comment]
Merged the PR, published version 4.1.0.
